app/consumer/msg_filter_ante.go

```diff
@@ -0,0 +1,45 @@
+package app
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	ibcconsumerkeeper "github.com/cosmos/interchain-security/x/ccv/consumer/keeper"
+)
+
+// MsgFilterDecorator defines an AnteHandler decorator that enables message
+// filtering based on certain criteria.
+type MsgFilterDecorator struct {
+	ConsumerKeeper ibcconsumerkeeper.Keeper
+}
+
+func NewMsgFilterDecorator(k ibcconsumerkeeper.Keeper) MsgFilterDecorator {
+	return MsgFilterDecorator{
+		ConsumerKeeper: k,
+	}
+}
+
+func (mfd MsgFilterDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
+	currHeight := ctx.BlockHeight()
+
+	// If the CCV channel has yet been established, then we must only allow certain
+	// message types.
+	if _, ok := mfd.ConsumerKeeper.GetProviderChannel(ctx); !ok {
+		if !hasValidMsgs(tx.GetMsgs()) {
+			return ctx, fmt.Errorf("tx contains unsupported message types at height %d", currHeight)
+		}
+	}
+
+	return next(ctx, tx, simulate)
+}
+
+func hasValidMsgs(msgs []sdk.Msg) bool {
+	for _, msg := range msgs {
+		msgType := sdk.MsgTypeURL(msg)
+		// TODO: Perform message filtering that only allows certain IBC messages
```

Comment:
@mpoke do you know what message types we need to support when the CCV channel has not been established yet?

Reply:
@alexanderbez All IBC messages. The client needs to be updated, the connection and the channel need to be set up, and then a normal IBC packet will arrive with valset updates, which will mark the CCV channel as established.

Besides, I don't see any other message that is necessary during channel initialization.